src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -525,3 +531,24 @@ def remove_missing_queries(self, widget_id, keep_ids):
 
 class DashboardSerializer(DashboardDetailsSerializer):
     title = serializers.CharField(required=True, max_length=255)
+
+
+def schedule_update_project_configs(dashboard: Dashboard):
+    """
+    Schedule a task to update project configs for all projects of an organization when a dashboard is updated.
+    """
+    org = dashboard.organization
+    projects = org.project_set.all()
+
+    on_demand_metrics = features.has("organizations:on-demand-metrics-extraction", org)
+    dashboard_on_demand_metrics = features.has(
+        "organizations:on-demand-metrics-extraction-experimental", org
+    )
+
+    if not projects or not on_demand_metrics or not dashboard_on_demand_metrics:
+        return
+
+    for project in projects:
+        schedule_invalidate_project_config(
+            trigger="dashboards:create-on-demand-metric", project_id=project.id
+        )
```

Comment:
Isn't that a bit excesive ? 
Could we maybe find out which projects have new on-demand metrics ?

Wouldn't this force an update for all projects in an organisation even if the modified widget had no on-demand-metrics?
To me it would seem normal to force an update only for the projects that are affected by widgets with on-demand metrics that have changed, this might not be so easy to do. 
At least we should only update the projects that actually have widgets with on-demand metrics on this dashboard regardless of whether they have changed.